src/D2L.CodeStyle.TestAnalyzers/Extensions/Microsoft.CodeAnalysis.cs

```diff
@@ -9,7 +9,7 @@ public static class RoslynExtensions {
 
 		// Copied from the non-test assembly because we do not reference it.
 
-		public static bool IsNullOrErrorType( this ITypeSymbol symbol ) {
+		public static bool IsNullOrErrorType( this ITypeSymbol? symbol ) {

```

Comment:
nit: could this be [annotated with `[NotNullWhen( false )]`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#conditional-post-conditions-notnullwhen-maybenullwhen-and-notnullifnotnull) to avoid needing the new method, or does that not work with extension methods?

Reply:
Oh interesting, I didn't realize it worked with non-`out` parameters (and indeed this case is right there in the docs)
I'll revisit it in this PR yep